services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,6 +231,34 @@ private void configureTaskActionClient(Binder binder)
                             .to(RemoteTaskActionClientFactory.class).in(LazySingleton.class);
 
           }
+
+          @Provides
+          @LazySingleton
+          public Task readTask(@Json ObjectMapper mapper, ExecutorLifecycleConfig config)
+          {
+            try {
+              return mapper.readValue(config.getTaskFile(), Task.class);
+            }
+            catch (IOException e) {
+              throw Throwables.propagate(e);
+            }
+          }
+
+          @Provides
+          @LazySingleton
+          @Named(MonitorsConfig.METRIC_DIMENSION_PREFIX + DruidMetrics.DATASOURCE)
```

Comment:
is MonitorsConfig.METRIC_DIMENSION_PREFIX necessary for the named injection? Since this field is not just use for metrics, it might make sense to give it a name that reflect that.